deta-play/deta_connection.py

```diff
@@ -1,14 +1,14 @@
 from streamlit.connections import ExperimentalBaseConnection
 from st_app_state.interface import StateBackend
 import deta
-from typing import Any, Union
+from typing import Union
 
 
 class DetaConnection(ExperimentalBaseConnection[deta.Base], StateBackend):
     """Basic st.experimental_connection implementation for Deta"""
 
     def _connect(self, **kwargs) -> deta.Base:
-        client = deta.Deta(self._secrets.data_key)
+        client = deta.Deta(self._secrets.deta_key)
```

Comment:
"Data Key" is actually what they call it in the UI so I was going off of that although it's confusing

![image](https://github.com/sfc-gh-jcarroll/llm-examples/assets/116604821/11178743-1083-483f-9e18-1bc94c0effb3)

Add [in the SDK it's a "project key"](https://github.com/deta/deta-python/blob/master/deta/__init__.py#L37)

Maybe it should just be `key=`? WDYT?

Reply:
or project key / project id / name / host would be all the params available to generate a Base currently (I can fix this in a separate PR and just merge this but wanted to discuss 😄 )

Reply:
Huh, I didn't realize that, I just thought it was a typo :). Maybe "api_key"? I think key or api_key seems most reasonable. data_key for a thing called deta is just too confusing :)

None of these changes is a must-have, just fiddling. I do like the global default key a lot, though :)

Reply:
Follow-up commit: https://github.com/sfc-gh-jcarroll/llm-examples/commit/53b04fe73f9b8f3734880e2596906a96a5bf57c6